contracts/Settlement/SettlementLogic.sol

```diff
@@ -446,52 +521,139 @@ contract SettlementLogic is ISettlement, SettlementStorage {
-                10**20
-            );
-            actualCollateralAmount = _collateralTokenSent.sub(relayerFee);
-        }
+        address _loanTokenAsset = ISovrynLoanToken(order.loanTokenAddress)
+            .loanTokenAddress();
+
+        if (order.loanTokenSent > 0 && order.collateralTokenSent > 0) {
+            (uint256 _rate, uint256 _precision) = IPriceFeeds(priceFeeds)
+                .queryRate(_loanTokenAsset, order.collateralTokenAddress);
+
+            uint256 _convertedLoanTokenSent = order
+                .loanTokenSent
+                .mul(_rate)
+                .div(_precision);
+
+            uint256 _orderSizeInColl = order.collateralTokenSent +
+                _convertedLoanTokenSent;
 
-        uint256 _loanTokenSent = order.loanTokenSent;
-        uint256 _feeLoanAssetByCollateral;
-        if (_loanTokenSent > 0) {
-            relayerFeeOnLoanAsset = _loanTokenSent.mul(relayerFeePercent).div(
-                10**20
+            relayerFee = _checkRelayerFee(
+                order.collateralTokenAddress,
+                _orderSizeInColl,
+                _orderSizeInColl,
+                false
             );
-            actualLoanTokenAmount = _loanTokenSent.sub(relayerFeeOnLoanAsset);
-            address loanTokenAsset = ISovrynLoanToken(order.loanTokenAddress)
-                .loanTokenAddress();
-            address[] memory _path = sovrynSwapNetwork.conversionPath(
-                loanTokenAsset,
-                order.collateralTokenAddress
+
+            if (relayerFee > order.collateralTokenSent) {
+                uint256 _outstandingFee = relayerFee -
+                    order.collateralTokenSent;
+                relayerFee = order.collateralTokenSent;
+                relayerFeeOnLoanAsset = _outstandingFee.mul(_precision).div(
+                    _rate
+                );
+            }
+        } else if (order.loanTokenSent > 0) {
+            relayerFeeOnLoanAsset = _checkRelayerFee(
+                _loanTokenAsset,
+                order.loanTokenSent,
+                order.loanTokenSent,
+                false
             );
-            _feeLoanAssetByCollateral = sovrynSwapNetwork.rateByPath(
-                _path,
-                relayerFeeOnLoanAsset
+        } else if (order.collateralTokenSent > 0) {
+            relayerFee = _checkRelayerFee(
+                order.collateralTokenAddress,
+                order.collateralTokenSent,
+                order.collateralTokenSent,
+                false
             );
+        } else {
+            revert("Invalid loanTokenSent and collateralTokenSent");
         }
 
-        _checkRelayerFee(
-            relayerFee + _feeLoanAssetByCollateral,
-            order.collateralTokenAddress
-        );
+        actualCollateralAmount = order.collateralTokenSent.sub(relayerFee);
+        actualLoanTokenAmount = order.loanTokenSent.sub(relayerFeeOnLoanAsset);
     }
 
```

Comment:
```
    function _checkAndGetRelayerFee(
```
makes it more clear what the function does

Reply:
Done: https://github.com/DistributedCollective/LimitOrder/commit/42bc5f345a822873960dea46a07dc26fc0585338